audit/hashstructure.go

```diff
@@ -134,6 +134,9 @@ type hashWalker struct {
 	// immediately and the error returned.
 	Callback HashCallback
 
+	// IgnoreKeys are the keys that wont have the HashCallback applied
+	IgnoredKeys []string
```

Comment:
I think this mechanism will have an issue: it won't apply keys only to the top level. I'm not sure whether that's _right_ or not but probably something that should be discussed. I think likely what will need to happen is if the top level key is not hmac'd anything under it shouldn't be, and vice versa. The problem of course is if you have something quite complex where you want to hash a value way down in the data but not at the top level, but I don't know offhand if we have any realistic scenarios like that, and I also can't think of a good way (other than dot indexing which would have its own issues) of tackling that...although that said, we could always add support for dot indexing at a later time.

Reply:
Yea, right now this will apply to all matched nested keys, and not top-level only keys. 

> I'm not sure whether that's right or not but probably something that should be discussed. 

I agree, probably worth discussing whether this is the behavior we want.

Reply:
Per discussion: we're not going to do this now but we could allow something like dot notation to specify. We could allow an additional parameter to replace dot if it's incompatible with someone's data, and for the top level if you just want that, you can have a trailing dot, e.g. `foo.` means only ignore `foo` at the top level, not further `foo`s.